### PR TITLE
Update: Sort dimension values in filter dropdown

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
The dimension select currently shows values directly as they are received from the backend

## Proposed Changes
- Sort the results so they are easier to scroll through
- add a check for sorting a list of string numbers to sort them numerically instead

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/filter-values/dimension-select.ts [hunk 1]
```diff
@@ -1,46 +1,57 @@
...
 
 interface DimensionSelectComponentArgs {
   filter: FilterFragment;
   request: RequestFragment;
   onUpdateFilter(changeSet: Partial<FilterFragment>): void;
 }
 
+function isNumeric(num: string) {
+  //@ts-ignore
+  return !isNaN(num);
```
Comment (jkusa) [inline packages/reports/addon/components/filter-values/dimension-select.ts:32]:
```suggestion
  return !isNaN(num as any);
```

File: packages/reports/addon/components/power-select-collection-options.ts [hunk 2]
```diff
@@ -4,13 +4,26 @@
...
+    groupKey?: string;
+    sortKey?: string;
+    sortFn?: (options: ItemOptions) => ItemOptions;
+  };
+};
+
+export type IndexedOptions<Option = ItemOptions[number]> = { option: Option; idx: number };
+
 export default class PowerSelectCollectionOptions extends Options {
+  declare args: Args;
```
Comment (jkusa) [inline packages/reports/addon/components/power-select-collection-options.ts:25]: Interesting, does something like this not work 
```
PowerSelectCollectionOptions<Args> extends Options
```
```
(PowerSelectCollectionOptions extends Options)<Args>

```

**Review by jkusa — APPROVED** [review]

File: packages/reports/addon/components/power-select-collection-options.ts (hunk 2 above)
Comment (kevinhinterlong) [reply packages/reports/addon/components/power-select-collection-options.ts:25]: Nope, easiest way is to override it like this unfortunately

File: packages/reports/addon/components/filter-values/dimension-select.ts (hunk 1 above)
Comment (kevinhinterlong) [reply packages/reports/addon/components/filter-values/dimension-select.ts]: Updated to use any instead of ts-ignore

**Pull request merged**